src/physfs_internal.h

```diff
@@ -69,7 +69,7 @@ extern "C" {
    All file-private symbols need to be marked "static".
    Everything shared between PhysicsFS sources needs to be in this
    file between the visibility pragma blocks. */
-#if PHYSFS_MINIMUM_GCC_VERSION(4,0) || defined(__clang__)
+#if !defined(_WIN32) && (PHYSFS_MINIMUM_GCC_VERSION(4,0) || defined(__clang__))
```

Comment:
Why `!defined(_WIN32)`? And MinGW?

Reply:
Because with _WIN32, thus MinGW included, dllexport attribute only causes the symbols marked with dllexport to be exported. visibility attributes result in lots of unsupported attribute warnings.

Reply:
Works like a charm with MinGW on my computer. Which warnings?

Reply:
Depends on the compiler version, possibly. Is the following good enough a reason for you?
```
src/physfs.c: In function '__PHYSFS_sort':
src/physfs.c:663:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_createMemoryIo':
src/physfs.c:411:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_createNativeIo':
src/physfs.c:238:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_strdup':
src/physfs.c:1433:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_hashString':
src/physfs.c:1442:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_getUserDir':
src/physfs.c:1656:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_readAll':
src/physfs.c:3143:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_initSmallAlloc':
src/physfs.c:3162:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_smallFree':
src/physfs.c:3175:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_DirTreeInit':
src/physfs.c:3257:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_DirTreeAdd':
src/physfs.c:3317:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_DirTreeFind':
src/physfs.c:3349:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_DirTreeEnumerate':
src/physfs.c:3372:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: In function '__PHYSFS_DirTreeDeinit':
src/physfs.c:3402:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs.c: At top level:
src/physfs.c:3402:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformDetectAvailableCDs':
src/physfs_platform_windows.c:422:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformCalcBaseDir':
src/physfs_platform_windows.c:506:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformCalcPrefDir':
src/physfs_platform_windows.c:545:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformCalcUserDir':
src/physfs_platform_windows.c:613:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformInit':
src/physfs_platform_windows.c:619:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformDeinit':
src/physfs_platform_windows.c:625:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformGetThreadID':
src/physfs_platform_windows.c:631:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformEnumerate':
src/physfs_platform_windows.c:696:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformMkDir':
src/physfs_platform_windows.c:708:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformOpenRead':
src/physfs_platform_windows.c:733:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformOpenWrite':
src/physfs_platform_windows.c:740:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformOpenAppend':
src/physfs_platform_windows.c:756:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformRead':
src/physfs_platform_windows.c:780:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformWrite':
src/physfs_platform_windows.c:805:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformSeek':
src/physfs_platform_windows.c:814:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformTell':
src/physfs_platform_windows.c:823:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformFileLength':
src/physfs_platform_windows.c:832:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformFlush':
src/physfs_platform_windows.c:840:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformClose':
src/physfs_platform_windows.c:847:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformDelete':
src/physfs_platform_windows.c:874:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformCreateMutex':
src/physfs_platform_windows.c:890:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformDestroyMutex':
src/physfs_platform_windows.c:897:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformGrabMutex':
src/physfs_platform_windows.c:904:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformReleaseMutex':
src/physfs_platform_windows.c:910:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_platform_windows.c: In function '__PHYSFS_platformStat':
src/physfs_platform_windows.c:1022:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_dir.c:193:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_unpacked.c: In function 'UNPK_closeArchive':
src/physfs_archiver_unpacked.c:54:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_unpacked.c: In function 'UNPK_abandonArchive':
src/physfs_archiver_unpacked.c:64:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_unpacked.c: In function 'UNPK_openRead':
src/physfs_archiver_unpacked.c:215:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_unpacked.c: In function 'UNPK_openWrite':
src/physfs_archiver_unpacked.c:221:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_unpacked.c: In function 'UNPK_openAppend':
src/physfs_archiver_unpacked.c:227:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_unpacked.c: In function 'UNPK_remove':
src/physfs_archiver_unpacked.c:233:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_unpacked.c: In function 'UNPK_mkdir':
src/physfs_archiver_unpacked.c:239:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_unpacked.c: In function 'UNPK_stat':
src/physfs_archiver_unpacked.c:266:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_unpacked.c: In function 'UNPK_addEntry':
src/physfs_archiver_unpacked.c:285:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_unpacked.c: In function 'UNPK_openArchive':
src/physfs_archiver_unpacked.c:302:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_grp.c:111:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_hog.c:165:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_7z.c: In function 'SZIP_global_init':
src/physfs_archiver_7z.c:406:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_7z.c: At top level:
src/physfs_archiver_7z.c:428:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_mvl.c:105:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_qpak.c:121:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_wad.c:130:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_zip.c:1705:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_slb.c:131:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_iso9660.c:381:1: warning: visibility attribute not supported in this configuration; ignored
src/physfs_archiver_vdf.c:164:1: warning: visibility attribute not supported in this configuration; ignored
```

Reply:
It's supported in recent MinGW. The proper way would be to rely on CMake to hide all symbols (`C_VISIBILITY_PRESET` property), it's more robust than a `#pragma GCC visibility push(hidden)` conditionally added.